### ISSUE TITLE
tutorial example breaks as soon as you add module-info.java

### ISSUE THREAD

**Issue by sdyura** [arangodb/arangodb-java-driver#577]
I downloaded https://github.com/arangodb/arangodb-java-driver/tree/main/tutorial/maven

and added my projects module-info.java, and the example stopped working

[arangodb-java-driver-demo.zip](https://github.com/user-attachments/files/17117454/arangodb-java-driver-demo.zip)

```
Exception in thread "main" com.arangodb.ArangoDBException: No JsonFactory found for content type: JSON
	at com.arangodb.serde.jackson.json@7.9.0/com.arangodb.serde.jackson.JacksonMapperProvider.of(JacksonMapperProvider.java:50)
	at com.arangodb.serde.jackson.json@7.9.0/com.arangodb.serde.jackson.JacksonSerde.of(JacksonSerde.java:26)
	at com.arangodb.serde.jackson.json@7.9.0/com.arangodb.serde.jackson.json.JacksonJsonSerdeProvider.create(JacksonJsonSerdeProvider.java:11)
	at com.arangodb.core@7.9.0/com.arangodb.internal.config.ArangoConfig.getUserDataSerde(ArangoConfig.java:265)
	at com.arangodb.core@7.9.0/com.arangodb.internal.config.ArangoConfig.getInternalSerde(ArangoConfig.java:271)
	at com.arangodb.core@7.9.0/com.arangodb.internal.net.Communication.<init>(Communication.java:36)
	at com.arangodb.http.HttpCommunication.<init>(HttpCommunication.java:39)
	at com.arangodb.http.HttpProtocolProvider.createProtocol(HttpProtocolProvider.java:33)
	at com.arangodb.core@7.9.0/com.arangodb.ArangoDB$Builder.build(ArangoDB.java:399)
	at net.yura.arango/net.yura.arango.FirstProject.main(FirstProject.java:25)
```

**Comment by rashtao** [comment]
Thanks for reporting, @sdyura 

This caused by https://github.com/FasterXML/jackson-core/issues/1340 and needs to be fixed and backported in Jackson core. 

For the moment, to work-around either:
- do not require Jackson in your `module-info.java` (and remove related code usages), or
- use the shaded Java driver  (`com.arangodb:arangodb-java-driver-shaded`) and the shaded Jackson classes (from package `com.arangodb.shaded.fasterxml.jackson`), or
- use the shaded Java driver (`com.arangodb:arangodb-java-driver-shaded`) and the Jackson JSON serde (`com.arangodb:jackson-serde-json`) and manually configure the JSON serde:
  ```java
        JsonFactory jf = new JsonFactory();
        JacksonUtils.tryConfigureJsonFactory(jf);
        ObjectMapper mapper = new ObjectMapper(jf);
        JacksonSerde serde = JacksonSerde.create(mapper);

        arangoDB = new ArangoDB.Builder()
                // ...
                .serde(serde)
                .build();
```

**Comment by rashtao** [comment]
Fixed in Jackson 2.18 and released in ArangoDB Java driver 7.10.0: https://github.com/arangodb/arangodb-java-driver/releases/tag/v7.10.0

Feel Free to reopen in case of further problems.
